install.sh

```diff
@@ -220,8 +220,7 @@ installconfig() {
 }
 
 check_usermode_support() {
-    user=$(systemctl --help|grep -e '--user')
-    [ -n "$user" ]
+    systemctl --help | grep -q -e '--user'
```

Comment:
What if systemctl itself fails? With the current code, it will trigger -e, with a pipe it doesn't, and the result of the query will be unintended.

We need to set pipefail too.

Reply:
#22385 has been merged.